src/server/tablet_snap.rs

```diff
@@ -805,6 +804,17 @@ where
             }
             Task::Send { addr, msg, cb } => {
                 let region_id = msg.get_region_id();
+                let to_peer_id = msg.get_to_peer().get_id();
+                defer!({
+                    if let Err(e) = self.snap_mgr.delete_snapshot(region_id, Some(to_peer_id)) {
```

Comment:
It's wrong to delete snapshot here while the sending task is being processed in the background.

Reply:
yes,  it should add in the limit and add defer in send_snap.